aws/code/workflows/MLWorkflowMultiGPU.py

```diff
@@ -0,0 +1,218 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import time
+
+import dask, dask_cudf
+from dask_cuda import LocalCUDACluster
+from dask.distributed import wait, Client
+
+import cupy
+import xgboost
+import joblib
+
+from dask_ml.model_selection import train_test_split
+from cuml.dask.common.utils import persist_across_workers
+from cuml.dask.ensemble import RandomForestClassifier
+from cuml.metrics import accuracy_score
+
+import warnings; warnings.filterwarnings("ignore")
+
+from MLWorkflow import MLWorkflow, timer_decorator
+
+class MLWorkflowMultiGPU ( MLWorkflow ):
+    """ Multi-GPU Workflow """
+
+    def __init__(self, hpo_config ):
+        print( 'Multi-GPU Workflow')
+        self.start_time = time.time()
+        
+        self.hpo_config = hpo_config
+        self.dataset_cache = None
+        
+        self.cv_fold_scores = []
+        self.best_score = -1
+
+        self.cluster, self.client = self.cluster_initialize()
+    
+    @timer_decorator
+    def cluster_initialize ( self ):
+        """ Initialize the dask compute cluster based on the number of available GPU workers.
+            XGBoost has a known issue where training fails if any worker has no data partition
+            so when initializing a dask cluster for xgboost we may need to limit the number of workers
+            see 3rd limitations bullet @ https://xgboost.readthedocs.io/en/latest/tutorials/dask.html 
+        """
+
+        cluster = None;  client = None
```

Comment:
Let's move these to separate lines. See the note on compound statements in https://www.python.org/dev/peps/pep-0008/#other-recommendations.

I noticed a few other compound statements earlier but didn't call them out because I'd forgotten that PEP8 officially discourages them. We might clean those up as well.